tidb-cloud/serverless-high-availability.md

```diff
@@ -0,0 +1,109 @@
+
+Applications remain unaffected by non-primary zone failures and are unaware of such issues. In the event of a primary zone failure, both Gateway and TiDB are launched in the standby zone to handle the workload. Applications should implement retry logic for connections to redirect new requests to the active servers in the standby zone.
+
+## Automatic Backups and Durability
+
+Database backups are a critical component of any business continuity and disaster recovery strategy as they help protect your data from corruption or accidental deletion. With these backups, you can restore your database to a specific point in time within the configured retention period, minimizing potential data loss and downtime.
+
+TiDB Cloud Serverless offers robust, automated backup mechanisms to ensure continuous protection of your data:
+
+- **Daily Full Backups**: A full backup of your database is taken once a day to capture the entire state of the database.
+- **Continuous Transaction Log Backups**: Transaction logs are backed up continuously, with backups occurring approximately every 5 minutes. The exact frequency may vary based on the level of database activity.
+
+These automated backups allow you to restore your database either from a full backup or from a specific point-in-time by combining full backups with continuous transaction logs. This flexibility ensures that you can recover your database to a precise point just before an incident occurred. 
+
+> **Note:**
+>
+> Automatic backups, including snapshot-based and continuous backups for Point-in-Time Recovery (PiTR), are performed on S3, which offers regional-level high durability.
+
+## Impact on Sessions During Failures
+
+In the event of a failure, ongoing transactions on the failed server are affected. Although failover remains transparent to your applications by default, applications need to implement logic to handle recoverable failures in the middle of a transaction. Here's what happens in various failure scenarios:
+
+- **TiDB Failures**: If a TiDB instance fails, the client connection remains unaffected because TiDB Cloud Serverless automatically re-routes traffic through the gateway. Transactions running on the failed TiDB instance may be interrupted, but the system ensures that committed data is not lost, and new transactions can be handled by another available TiDB instance.
+- **Gateway Failures**: If the Gateway fails, it disrupts the client connection. TiDB Cloud Serverless gateways are stateless, allowing them to be restarted immediately in a new zone or server. The system also automatically redirects traffic to the new gateway, minimizing downtime.
```

Comment:
```suggestion
- **TiDB failures**: If a TiDB instance fails, client connections are unaffected because TiDB Cloud Serverless automatically reroutes traffic through the gateway. While transactions on the failed TiDB instance might be interrupted, the system ensures that committed data is preserved, and new transactions are handled by another available TiDB instance.
- **Gateway failures**: If the Gateway fails, client connections are disrupted. However, TiDB Cloud Serverless gateways are stateless and can restart immediately in a new zone or server. Traffic is automatically redirected to the new gateway, minimizing downtime.
```